sql/core/src/main/scala/org/apache/spark/sql/execution/command/tables.scala

```diff
@@ -538,12 +539,25 @@ case class TruncateTableCommand(
               }
             }
             optAcls.foreach { acls =>
+              val aclEntries = acls.asScala.filter(_.getName != null).asJava
```

Comment:
Hi, @viirya . So, does your cluster has no default ACL configuration in production? I'm wondering if that is intentional or not.

Reply:
Usually, `directory` has default ACL configuration and the new directory inherits from the parent and builds its own default ACL, doesn't it?

Reply:
I'm not sure. For the failed case of our customer, there is no default ACLs so causing this issue. I think we should not assume it always has it.

Reply:
Ya. The approach is correct for that case. I was wondering the default ACL is removed by the user or not in that cluster.